File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-14.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        se=set()
        i=j=0
        res=0
        while j<len(s):
            while s[j] in se:
                se.remove(s[i])
                i+=1
            se.add(s[j])
            res=max(res,j-i+1)
            j+=1
        return res
            

File: Data Structures & Algorithms/sliding-window-maximum/submission-14.java
class Solution {
    public int[] maxSlidingWindow(int[] nums, int k) {
        int []res=new int[nums.length-k+1];
        Deque<Integer> deque=new LinkedList<>();
        for(int i=0;i<nums.length;i++){
            if(!deque.isEmpty() && i-k+1>deque.peek()) deque.pollFirst();
            while(!deque.isEmpty() && nums[deque.peekLast()]<nums[i]) deque.pollLast();
            deque.offer(i);
            if(i>=k-1)res[i-k+1]=nums[deque.peek()];

        }
        return res;
    }
}
